src/components/SearchBox/SearchBox.js

```diff
@@ -72,13 +73,18 @@ class SearchBox extends Component {
 
     if (searchAsYouType) {
       refine(query);
-    } else {
-      this.setState({ query });
     }
+    this.setState({ query });
 
     onChange(event);
   };
 
+  componentWillReceiveProps(nextProps) {
+    if (!this.state.focused && nextProps.query !== this.state.query) {
```

Comment:
We should add an explanation that mentions why this branch is necessary (because of the `queryHook` option). This is not straightforward to understand.

Reply:
agreed, it's done here: e205c2169766de15b853f3772edd7c674855510e